com.unity.ml-agents/Runtime/Inference/BarracudaModelParamLoader.cs

```diff
@@ -8,13 +8,29 @@
 
 namespace Unity.MLAgents.Inference
 {
+
     /// <summary>
     /// Prepares the Tensors for the Learning Brain and exposes a list of failed checks if Model
     /// and BrainParameters are incompatible.
     /// </summary>
     internal class BarracudaModelParamLoader
     {
-        const long k_ApiVersion = 2;
+
+        internal enum CheckType
+        {
+            Info = 0,
+            Warning = 1,
+            Error = 2
+        }
+
+        internal class FailedCheck
+        {
+            public CheckType CheckType;
+            public string Message;
```

Comment:
How about adding a `static FailedCheck Info(string message)` helper method (and same for Warning and Error)? That would simplify a bit of the checking code.